input/drivers/android_input.c

```diff
@@ -869,7 +871,7 @@ static void handle_hotplug(android_input_data_t *android_data,
    if (!back_mapped && settings->input.back_as_menu_toggle_enable)
       settings->input.autoconf_binds[*port][RARCH_MENU_TOGGLE].joykey = AKEYCODE_BACK;
 
-   android_data->pad_states[android_data->pads_connected].id = id;
+   android_data->pad_states[*port].id = id;
```

Comment:
As you said earlier, this is not a something to simply merge before considering and testing the suggestion. I'm sorry that I wrote that this was for Twinaphex, if you are actually the one working on the input system.

About the changing of index for android_data->pad_states[]:
I'm actually thinking that is an error that *port is not being used as index in the two lines. I should have corrected both lines in my pull request. Shouldn't always be *port that is used as index, since this is actually the index you are modifying the controller on? If you look at the strlcpy two lines further below, it is using *port as index as well.

When would it make sense to use android_data->pads_connected as index when it does not match the value of *port?